File: 1_two_sum/1_two_sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        //int[] result =new int[2];
        for(int i=0; i<nums.length;i++){
               for(int j=i+1; j<nums.length;j++){
                   //int temp = nums[i]+nums[j];
                if(nums[j] == target - nums[i]){
                    //result[0] = nums[i];
                    //result[1] = nums[j];
                    return new int[] {i,j};
                }
            } 
        }
        throw new IllegalArgumentException("No two sum solution");
    }
}

File: 1_two_sum/1_two_sum.go
package __two_sum

import "fmt"

func main() {
	nums := []int{2, 7, 11, 15}
	target := 9
	fmt.Printf("nums: %v\t target: %v\t twoSum: %v\n", nums, target, twoSum(nums, target))
}

func twoSum(nums []int, target int) []int {
	length := len(nums)
	a := map[int]int{}
	for i := 0; i < length; i++ {
		if v, ok := a[target-nums[i]]; ok {
			return []int{v, i}
		}
		a[nums[i]] = i
	}
	return []int{}

}
